includes/multi-currency/Compatibility/WooCommerceDeposits.php

```diff
@@ -0,0 +1,151 @@
+	 * @var Utils
+	 */
+	private $utils;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param MultiCurrency $multi_currency MultiCurrency class.
+	 * @param Utils         $utils Utils class.
+	 */
+	public function __construct( MultiCurrency $multi_currency, Utils $utils ) {
+		$this->multi_currency = $multi_currency;
+		$this->utils          = $utils;
+		$this->initialize_hooks();
+	}
+
+	/**
+	 * Adds compatibility filters if the plugin exists and loaded
+	 *
+	 * @return  void
+	 */
+	protected function initialize_hooks() {
+		if ( class_exists( 'WC_Deposits' ) ) {
+			// Add compatibility filters here.
+			add_action( 'woocommerce_deposits_create_order', [ $this, 'modify_order_currency' ] );
+			add_filter( 'woocommerce_get_cart_contents', [ $this, 'modify_cart_item_deposit_amounts' ] );
+			add_filter( 'woocommerce_product_get__wc_deposit_amount', [ $this, 'modify_cart_item_deposit_amount_meta' ], 10, 2 );
+			add_filter( MultiCurrency::FILTER_PREFIX . 'should_convert_product_price', [ $this, 'maybe_convert_product_prices_for_deposits' ], 10, 2 );
+		}
+	}
+
+	/**
+	 * Converts the currency for deposit amounts of each cart item, only applies if deposits are enabled on the product.
+	 *
+	 * @param array $cart_contents The current tax definitions.
+	 *
+	 * @return array $tax Array of altered taxes.
+	 */
+	public function modify_cart_item_deposit_amounts( $cart_contents ) {
+		foreach ( $cart_contents as $cart_item_key => $cart_item ) {
+			if ( isset( $cart_item['is_deposit'] ) && $cart_item['is_deposit'] && isset( $cart_item['deposit_amount'] ) ) {
+					$deposit_amount                                    = floatval( $cart_item['deposit_amount'] );
+					$cart_contents[ $cart_item_key ]['deposit_amount'] = $this->multi_currency->get_price( $deposit_amount, 'product' );
+			}
+		}
+
+		return $cart_contents;
+	}
+
+	/**
+	 * Converts the currency for deposit amounts of each cart item, only applies if deposits are enabled on the product.
+	 *
+	 * @param array $cart_contents The current tax definitions.
+	 *
+	 * @return array $tax Array of altered taxes.
+	 */
+	public function modify_cart_item_deposit_amount_meta( $amount, $product ) {
+		if ( $this->utils->is_call_in_backtrace( [ 'WC_Deposits_Cart_Manager->deposits_form_output' ] )
+		&& 'percent' !== WC_Deposits_Product_Manager::get_deposit_type( $product ) ) {
+			return $this->multi_currency->get_price( $amount, 'product' );
+		}
+		return $amount;
+	}
+
+	/**
+	 * Defines if the product prices need to be converted when calculating totals,
+	 * if the product's deposit type is a payment plan, then it shouldn't convert it.
+	 *
+	 * @param   bool        $result   The previous flag for converting the price.
+	 * @param   \WC_Product $product  The product to check for.
+	 *
+	 * @return  bool                        Whether the price should be converted or not.
+	 */
+	public function maybe_convert_product_prices_for_deposits( $result, $product ) {
+		if ( class_exists( 'WC_Deposits_Product_Manager' )
```

Comment:
I think it's safe _not_ to test to see if `WC_Deposits_Product_Manager` exists like in `modify_cart_item_deposit_amount_meta` due to this should only be called if Deposits is installed and active. I don't think the Deposits extension will have a large enough change to remove that class or method.

Reply:
Yes, psalm also agrees that :)